### DISCUSSION TITLE
Minio has any different in comparison with one year ago when store large number of small files.

### DISCUSSION THREAD

**Discussion by meotimdihia** [minio/minio#16334] [Q&A] [answered]
I used Minio one year ago to store 30TB images, ~50 million images.
With HDD storage, Minio's performance was worse than Seaweedfs.
Now, I can't upgrade Seaweedfs to a newer version because of bugs. I am thinking about switching back to Minio.
I want to ask If there are changes to this problem.

**Comment by harshavardhana** [comment, marked as answer]
Many things have improved dramatically, 50 million images is a small-scale use case for us. We have deployments with 3billion+ small objects.

Of course for small objects you are better off using NVMe, it shall give you the much-required performance and latency.

HDDs are poor at handling small blocks so you should not expect NVMe level performance. However, you will see dramatic improvements even on HDDs some articles talk about these changes in some detail.

- https://blog.min.io/minio-versioning-metadata-deep-dive/
- https://blog.min.io/minio-optimizes-small-objects/
- https://blog.min.io/small_objects/

**Reply by meotimdihia** [reply]
Thanks for the answer.
